dsn_test.go

```diff
@@ -236,16 +236,30 @@ func TestParseDSN(t *testing.T) {
 			dsn: fmt.Sprintf("u:p@ac.snowflake.local:9876?account=ac&protocol=http&authenticator=SNOWFLAKE_JWT&privateKey=%v", privKeyPKCS8),
 			config: &Config{
 				Account: "ac", User: "u", Password: "p",
-				Authenticator: authenticatorJWT, PrivateKey: testPrivKey,
+				AuthType: AuthTypeJwt, PrivateKey: testPrivKey,
 				Protocol: "http", Host: "snowflake.local", Port: 9876,
 			},
 			err: nil,
 		},
+		{
+			dsn: fmt.Sprintf("u:p@ac.snowflake.local:9876?account=ac&protocol=http&authenticator=%v", url.QueryEscape("https://ac.okta.com")),
```

Comment:
This is the test case to make sure the authenticator type is passed in correctly though dsn